src/sentry/rules/conditions/event_frequency.py

```diff
@@ -369,19 +378,52 @@ class EventUniqueUserFrequencyCondition(BaseEventFrequencyCondition):
     def query_hook(
         self, event: GroupEvent, start: datetime, end: datetime, environment_id: str
     ) -> int:
-        totals: Mapping[int, int] = self.tsdb.get_distinct_counts_totals(
-            model=get_issue_tsdb_user_group_model(event.group.issue_category),
+        totals: Mapping[int, int] = self.get_snuba_query_result(
+            tsdb_function=self.tsdb.get_distinct_counts_totals,
             keys=[event.group_id],
+            group=event.group,
+            model=get_issue_tsdb_user_group_model(event.group.issue_category),
             start=start,
             end=end,
             environment_id=environment_id,
-            use_cache=True,
-            jitter_value=event.group_id,
-            tenant_ids={"organization_id": event.group.project.organization_id},
             referrer_suffix="alert_event_uniq_user_frequency",
         )
         return totals[event.group_id]
 
+    def batch_query_hook(
+        self, group_ids: Sequence[int], start: datetime, end: datetime, environment_id: str
+    ) -> dict[int, int]:
+        batch_totals: dict[int, int] = defaultdict(int)
+        groups = Group.objects.filter(id__in=group_ids)
+        error_issues = [group for group in groups if group.issue_category == GroupCategory.ERROR]
+        generic_issues = [group for group in groups if group.issue_category != GroupCategory.ERROR]
```

Comment:
nit: thoughts on making a helper for this to return a tuple? since we're doing it in a few places might be nice to simplify the logic and put it all in 1 spot. That will also allow us to make changes to the logic in 1 spot if we need to split out more issue types etc.